File: backend/internal/constants/attribution_state.go
package constants

type AttributionState string

const (
	AttributionQueued      AttributionState = "queued"
	AttributionCalculating AttributionState = "calculating"
	AttributionCompleted   AttributionState = "completed"
	AttributionFailed      AttributionState = "failed"
	AttributionReviewed    AttributionState = "reviewed"
	AttributionConfirmed   AttributionState = "confirmed"
	AttributionVoided      AttributionState = "voided"
)

var AttributionTransitions = map[AttributionState]map[AttributionState]bool{
	AttributionQueued:      {AttributionCalculating: true},
	AttributionCalculating: {AttributionCompleted: true, AttributionFailed: true, AttributionReviewed: true},
	AttributionCompleted:   {AttributionReviewed: true, AttributionVoided: true},
	AttributionFailed:      {AttributionCompleted: true},
	AttributionReviewed:    {AttributionConfirmed: true, AttributionVoided: true},
	AttributionConfirmed:   {AttributionCompleted: true},
	AttributionVoided:      {AttributionCompleted: true},
}

func CanTransitionAttribution(from, to AttributionState) bool {
	return AttributionTransitions[from][to]
}

File: frontend/src/types/enums/attribution-state.ts
export type AttributionState = 'queued' | 'calculating' | 'completed' | 'failed' | 'reviewed' | 'confirmed' | 'voided'

export const attributionStateLabels: Record<AttributionState, string> = {
  queued: '排队中', calculating: '计算中', completed: '待复核', failed: '失败',
  reviewed: '已复核', confirmed: '已确认', voided: '已作废',
}
